nvflare/app_opt/xgboost/histogram_based_v2/adaptors/grpc_server_adaptor.py

```diff
@@ -132,7 +132,7 @@ def start(self, fl_ctx: FLContext):
         if not port:
             raise RuntimeError("failed to get a port for XGB server")
 
-        server_addr = f"localhost:{port}"
+        server_addr = f"127.0.0.1:{port}"
```

Comment:
@nvidianz @IsaacYangSLA mentioned something about localhost is not on Azure, so this is a good change.